### DISCUSSION TITLE
Get users list of a tenant with some additional details

### DISCUSSION THREAD

**Discussion by nidhikachhawa24** [Sunbird-Ed/Community#391] [Sunbird ED configuration discussion] [answered]
Hello all,

I am looking for a **user search API** of a tenant where I can get
- users roles ("PUBLIC" , "CONTENT_CREATOR", etc) and 
- their sub-profiles.

Also, how we can extend the limit of users in search API? 
Through **/api/user/v1/search** API we are able to see only 10000 users (What if a tenant is having 10000+ users)

cc: @rajeevsathish @vinukumar-vs @Krishnaj20 @reshmi-nair @rhwarrier @coolbung @baljeetsingh

**Comment by coolbung** [comment]
If you search by name / email you would be able to filter it out to a smaller list right ? What is the use case for fetching more than 10k records ?

**Reply by nidhikachhawa24** [reply]
By default we are getting 250 users list. By setting limit we can get 10000 users.

**Use case:** We want to check the contents created by those users

**Comment by vinukumar-vs** [comment]
@reshmi-nair is it configurable? If not any specific reason for restriction? 
 
 @gandham-santhosh

**Reply by reshmi-nair** [reply]
@nidhikachhawa24 @vinukumar-vs This is not configurable currently from Lern side. 10000 is the default setting by ES and we are using that. 
@nidhikachhawa24  Is the above mentioned use case for reporting or display?

**Reply by chitranshu-keshav** [reply]
@nidhikachhawa24 incase your query answered can you please mark this thread as answered otherwise please get back with the required details

**Reply by nidhikachhawa24** [reply]
@reshmi-nair The above case is for displaying the users in a dashboard. Suppose I've 20,000 users in a specific tenant and want to display all the users in the UI then how am I suppose to achieve it if the limit is only 10000?

**Reply by rhwarrier** [reply]
Not sure how displaying 20k users in one dashboard will be of use to anybody - however, if that is what the use case demands, maybe there are ways to make multiple calls. Please tag @amiableanil  https://github.com/AmiableAnil  for help with queries going forward. Thanks

**Reply by nidhikachhawa24** [reply]
Hello @AmiableAnil can you help me in this?

**Reply by AmiableAnil** [reply, marked as answer]
@nidhikachhawa24 Displaying the 20K users is not a good design and we are also not supporting is as of now. Ideally we should narrow down the search result by putting the more filters to it. This is not even suggested by ElasticSearch. Please find the ElasticSearch thread here https://discuss.elastic.co/t/pulling-more-than-10000-records-from-elasticsearch-query/181000/1.
